src/vm/jitinterface.cpp

```diff
@@ -6159,11 +6172,31 @@ CorInfoHelpFunc CEEInfo::getNewArrHelperStatic(TypeHandle clsHnd)
     if (CorTypeInfo::IsGenericVariable(elemType))
     {
         result = CORINFO_HELP_NEWARR_1_OBJ;
+        *pHasSideEffects = true;
     }
     else if (CorTypeInfo::IsObjRef(elemType))
     {
         // It is an array of object refs
         result = CORINFO_HELP_NEWARR_1_OBJ;
+        if (thElemType.IsTypeDesc())
+        {
+            *pHasSideEffects = true;
+        }
+        else
+        {
+            MethodTable * pElemMT = thElemType.AsMethodTable();
+            bool hasFinalizer = pElemMT->HasFinalizer();
```

Comment:
This does not look right or necessary. There are no arrays with finalizers.

The potential side-effects from array allocations come from size. People may allocate very large array and expect it throw OutOfMemory exceptions. What does the JIT do in this case today?

Reply:
You are correct, getNewArrHelper were not needed, I removed them. The jit will never remove calls to array allocation helpers since they have noThrow set to false.